Google-Summer-of-Code-2023.md

```diff
@@ -697,6 +699,44 @@ Additionally, finish adding [computer-aided translation (CAT)](https://docs.goog
 **Useful resources:**
 
 
+### 2.3. Improvements for translation reviewer experience
+
+**Project Description:**
+
+Translation reviewers have given feedback such that the current contributor dashboard reviewer workflow doesn’t match with most of their needs. This project aims to improve the translation reviewer experience in the contributor dashboard.
+
+**Size of this project:** large (~350 hours)
+
+**Potential Mentors:** TBA
+
+**Knowledge/Skills Recommended:**
+
+* Knowledge and understanding of Python
+* Knowledge and understanding of TypeScript, Angular, basic HTML, and CSS
+
+**Suggested Milestones:**
+
+* **Milestone 1:**
+  - Showing translation cards according to the order of the content flow in the lesson.
+  - Exploration editor should also be linked so that the original lesson can be seen.
+  - Showing the size of the translation(short/large) ([Mocks in progress](https://github.com/oppia/design-team/issues/53))
+  - Pin/unpin specific lessons to translate ([Mocks in progress](https://github.com/oppia/design-team/issues/53))
+
+
+* **Milestone 2:**
+  - Ability to revert the acceptance or rejection of cards. [The exact time limit that we allow to undo the review should be decided](https://docs.google.com/document/d/1lIXkcQwPcPeS715vhX6mY-lowzltLxE8v2QQEvFadks/edit?disco=AAAAn56QFwE).
+  - Showing image alt text below the image. ([Mocks in progress](https://github.com/oppia/design-team/issues/54))
+  - Notifying new submissions through emails. [Existing CD reviewer notification system](https://docs.google.com/document/d/1tuXPSHvUN6I32Rle7E2CQs5HONgtP3TMuA__bWEsoDY/edit#heading=h.qzrvkvw31j3i).
+
+**Dependency on Release Schedule:** 
+
+**Proposal notes:**
+
+* To support lesson pinning, we have to make sure the pinned lessons are returned at the top of the list of lessons with translation suggestions in the fetch query [here](https://github.com/oppia/oppia/blob/develop/core/controllers/contributor_dashboard.py#L336) (`_get_reviewable_exploration_opportunity_summaries()`).
+* We will likely want to introduce a new storage model to track pinned lessons by language and user.
```

Comment:
@sagangwee I suggest adding some notes on:
- "Showing translation cards according to the order of the content flow in the lesson." -- how will this content flow be determined
- For the storage model -- suggest giving the name of the model, the ID, and the fields. (Is there already any per-user model describing their interactions with the CD that we can tack this info onto?)
- "Ability to revert the acceptance or rejection of cards." -- I feel like we need to resolve this with @dchen97  more conclusively before we can offer this as part of the project. The requirements aren't clear yet and that will likely affect the technical approach...

Reply:
waiting for response from @sagangwee

Reply:
Re the first 2 bullets, updated the technical hints section of the doc: https://docs.google.com/document/d/1lEEAJC9kKvOJ37zw-LLTsBS9uxsfXwjaCZDOWE_gubA/edit#.

Re the undo feature, I believe some time last week we settled on 30 seconds. Probably should read "Ability to undo the acceptance or rejection of a translation card for up to 30 seconds." Also added a note on potentially using a task queue for this in the doc.